### PR TITLE
Clean domain method

### PR DESCRIPTION
**Pull request by ebrillhart**
this should help fix https://github.com/FormidableLabs/victory-chart/issues/146, #306, #299 

right now this works great for everything except bar scales, which is weird because it's working with area and that's the same domain method. might be something to look at today in pairing @boygirl

### CODE REVIEW

File: src/helpers/domain.js [hunk 1]
```diff
@@ -15,7 +16,30 @@ export default {
     }
     const dataset = Data.getData(props);
     const domain = this.getDomainFromData(props, axis, dataset);
-    return this.padDomain(domain, props, axis);
+    return this.cleanDomain(this.padDomain(domain, props, axis), props);
+  },
+
+  cleanDomain(domain, props) {
```
Comment (boygirl) [inline src/helpers/domain.js]: I think you can simplify this method if you add an `axis` argument

File: src/helpers/domain.js [hunk 2]
```diff
@@ -15,7 +16,30 @@ export default {
...
+      y: Scale.getScaleType(props, "y")
+    };
+
+    if (scaleType.x !== "log" && scaleType.y !== "log") {
+      return domain;
+    }
+
+    const rules = (dom, axis) => {
+      const domainOne = dom[0] === 0 ? 1 / Number.MAX_SAFE_INTEGER : dom[0];
+      const domainTwo = dom[1] === 0 ? -Math.abs(1 / Number.MAX_SAFE_INTEGER) : dom[1];
```
Comment (boygirl) [inline src/helpers/domain.js]: Be careful with the logic around having one of these be negative and the other positive. The domain is not always `[min, max]`

**Comment by boygirl** [comment]
cc/ @ebrillhart I figured it out! We needed to replace the baseline (y0) with a very small number in these charts prior to scaling them.

**Comment by ebrillhart** [comment]
@boygirl yay!! what else needs to be done on this?

**Comment by boygirl** [comment]
@ebrillhart the y-axis labels (and possibly domain) on log scales are wrong for bar and area. You can address that in this branch or open another issue. Your pick.

**Comment by boygirl** [comment]
@ebrillhart actually pause. I think I messed something up with horizontal bars. One moment

**Comment by boygirl** [comment]
Okay, fixed. Sorry about that

**Comment by ebrillhart** [comment]
@boygirl no prob! was the weird tick label and domain thing the aspect that we determined was a d3 issue? what would your suggestion be for a starting approach to that?

**Comment by boygirl** [comment]
@ebrillhart please check on what's getting passed to VictoryAxis in these cases. If it's getting the domain and scale that you expect then I think it's a d3 issue with tick formatting, and we will address it in it separate issue.

**Pull request merged**